src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java

```diff
@@ -29,28 +29,44 @@
 import hudson.model.AbstractBuild;
 import hudson.model.AbstractProject;
 import hudson.model.BuildListener;
+import hudson.model.Run.RunnerAbortedException;
 import hudson.tasks.BuildWrapper;
 import hudson.tasks.BuildWrapperDescriptor;
 import java.io.IOException;
+import java.io.OutputStream;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import org.jenkinsci.plugins.credentialsbinding.MultiBinding;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 @SuppressWarnings({"rawtypes", "unchecked"}) // inherited from BuildWrapper
 public class SecretBuildWrapper extends BuildWrapper {
 
     private final List<? extends MultiBinding<?>> bindings;
+    private boolean showBindings;
 
+    public SecretBuildWrapper(List<? extends MultiBinding<?>> bindings, boolean showBindings) {
+        this.bindings = bindings;
+        this.showBindings = showBindings;
+    }
+    
     @DataBoundConstructor public SecretBuildWrapper(List<? extends MultiBinding<?>> bindings) {
         this.bindings = bindings;
+        showBindings = true;
```

Comment:
Why? The setter will be called immediately afterwards anyway.

And if you meant to retain former behavior for projects loaded from disk for some reason (I do not recommend it), this does not do it. You would need to add a `readResolve`.